odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/DataSourceRepositoryImpl.java

```diff
@@ -302,14 +328,26 @@ private Long fetchCount(final String query) {
     private DataSourceDto mapRecord(final Record record, final String dataSourceCteName) {
         return new DataSourceDto(
             jooqRecordHelper.remapCte(record, dataSourceCteName, DATA_SOURCE).into(DataSourcePojo.class),
-            record.into(NAMESPACE).into(NamespacePojo.class)
+            record.into(NAMESPACE).into(NamespacePojo.class),
+            record.into(TOKEN).map((map) -> {
+                if (map.get(TOKEN.UPDATED_AT).isBefore(LocalDateTime.now().minusSeconds(1))) {
```

Comment:
let's fix this logic and return token as is for create request and hide it on all get requests